File: VECTOAux/VectoAuxiliaries/Electrics/CombinedAlternatorSignals.vb

Imports TUGraz.VectoCommon.Utils

Namespace Electrics
	'Used by the CombinedAlternator class and any other related classes.
	Public Class CombinedAlternatorSignals
		Implements ICombinedAlternatorSignals

		Public Property CrankRPM As Double Implements ICombinedAlternatorSignals.CrankRPM

		Public Property CurrentDemandAmps As Ampere Implements ICombinedAlternatorSignals.CurrentDemandAmps

		'Number of alternators in the Combined Alternator
		Public Property NumberOfAlternators As Integer Implements ICombinedAlternatorSignals.NumberOfAlternators
	End Class
End Namespace

File: VECTOAux/BusAuxiliaries/DownstreamModules/Impl/Electrics/CombinedAlternatorSignals.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using DownstreamModules.Electrics;
using TUGraz.VectoCommon.Utils;

namespace Electrics
{
	// Used by the CombinedAlternator class and any other related classes.
	public class CombinedAlternatorSignals : ICombinedAlternatorSignals
	{
		public double CrankRPM { get; set; }

		public Ampere CurrentDemandAmps { get; set; }

		// Number of alternators in the Combined Alternator
		public int NumberOfAlternators { get; set; }
	}
}
